src/state/Connection.ts

```diff
@@ -250,6 +250,20 @@ export class Connection {
       [],
     );
 
+    scope
```

Comment:
maybe we could have a switch map from a new startLifecycle observable if you prefer